packages/service-provider-core/src/uri-generator.ts

```diff
@@ -129,8 +129,8 @@ function generateUri(options: CliOptions): string {
     return '';
   }
   const connectionString = generateUriNormalized(options);
-  if (connectionString.hosts.length === 1 &&
-      ['localhost', '127.0.0.1'].includes(connectionString.hosts[0].split(':')[0])) {
+  if (connectionString.hosts.every(host =>
+    ['localhost', '127.0.0.1'].includes(host.split(':')[0]))) {
```

Comment:
This is more of a drive-by change to account for the fact that the e2e-direct test cannot use serverSelectionTimeoutMS directly anymore, but I feel like it still just makes sense